### PR TITLE
Install latest patch versions of golang for linux

### PR DESCRIPTION
**Pull request by dmitry-shibanov**
- install latest patch versions for golang
#63 
#458

### CODE REVIEW

File: images/linux/scripts/installers/go.sh [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+golangTags="/tmp/golang_tags.json"
+# This function installs Go using the specified arguments:
+#   $1=MajorVersion (1.11)
+#   $2=IsDefaultVersion (true or false)
+function InstallGo () {
+    version=$( getFullGoVersion $1 )
+    echo "Install Go $version"
+    curl -sL https://dl.google.com/go/$version.linux-amd64.tar.gz -o $version.linux-amd64.tar.gz
+    mkdir -p /usr/local/go$1
+    tar -C /usr/local/go$1 -xzf $version.linux-amd64.tar.gz --strip-components=1 go
```
Comment (AlenaSviridenko) [inline images/linux/scripts/installers/go.sh]: can we create a variables for `/usr/local/go$1` and `$version.linux-amd64.tar.gz`? it is repeated multiple times in this function and creates some mess with all these slashes and dots. Something like this look a bit more better I think
```
    version=$( getFullGoVersion $1 )
    downloadVersion=$version.linux-amd64.tar.gz
    goFolder=/usr/local/go$1

    echo "Install Go $version"
    curl -sL https://dl.google.com/go/$downloadVersion -o $downloadVersion
    mkdir -p $goFolder
    tar -C $goFolder -xzf $downloadVersion --strip-components=1 go
    rm $downloadVersion
    echo "GOROOT_${1//./_}_X64=$goFolder" | tee -a /etc/environment
    DocumentInstalledItem "Go $1 ($($goFolder/bin/go version))"

    # If this version of Go is to be the default version,
    # symlink it into the path and point GOROOT to it.
    if [ $2 = true ]
    then
        ln -s $goFolder/bin/* /usr/bin/
        echo "GOROOT=$goFolder" | tee -a /etc/environment
    fi
```

File: images/linux/scripts/installers/go.sh [hunk 2]
```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+################################################################################
+##  File:  go.sh
+##  Desc:  Installs go, configures GOROOT, and adds go to the path
+################################################################################
+
+# Source the helpers for use with the script
+source $HELPER_SCRIPTS/document.sh
+golangTags="/tmp/golang_tags.json"
```
Comment (maxim-lobanov) [inline images/linux/scripts/installers/go.sh]: nitpick: enter between 9 and 10 lines for readability

File: images/linux/scripts/installers/go.sh [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+        echo "GOROOT=/usr/local/go$1" | tee -a /etc/environment
+    fi
+}
+
+function getFullGoVersion () {
+    local pattern="refs/tags/go$1([.0-9]{0,3})$"
+    local query='[.[] | select( .ref | test($pattern))] | .[-1] | .ref'
+    refValue=$(jq --arg pattern "$pattern" "$query" $golangTags)
+    version=$(echo "$refValue" | cut -d '/' -f 3)
+    version=$( echo "${version//\"}" ) # go1.12.17
```
Comment (maxim-lobanov) [inline images/linux/scripts/installers/go.sh]: nitpick: space

File: images/linux/scripts/installers/go.sh [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    echo $version
+}
+
+# load golang_tags.json file
+curl -s 'https://api.github.com/repos/golang/go/git/refs/tags' >> $golangTags
+echo "log go version"
```
Comment (maxim-lobanov) [inline images/linux/scripts/installers/go.sh]: please remove debug lines

File: images/linux/scripts/installers/go.sh [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+# Source the helpers for use with the script
+source $HELPER_SCRIPTS/document.sh
+golangTags="/tmp/golang_tags.json"
+# This function installs Go using the specified arguments:
+#   $1=MajorVersion (1.11)
+#   $2=IsDefaultVersion (true or false)
+function InstallGo () {
+    version=$( getFullGoVersion $1 )
+    echo "Install Go $version"
+    curl -sL https://dl.google.com/go/$version.linux-amd64.tar.gz -o $version.linux-amd64.tar.gz
```
Comment (maxim-lobanov) [inline images/linux/scripts/installers/go.sh]: `curl -sL https://dl.google.com/go/$version.linux-amd64.tar.gz`
I would suggest to use `${version}` approach in such lines: `curl -sL https://dl.google.com/go/${version}.linux-amd64.tar.gz` for readability

File: images/linux/scripts/installers/go.sh (hunk 1 above)
Comment (dmitry-shibanov) [reply images/linux/scripts/installers/go.sh]: Done.

File: images/linux/scripts/installers/go.sh (hunk 4 above)
Comment (dmitry-shibanov) [reply images/linux/scripts/installers/go.sh]: Done.

File: images/linux/scripts/installers/go.sh (hunk 3 above)
Comment (dmitry-shibanov) [reply images/linux/scripts/installers/go.sh]: Done.

File: images/linux/scripts/installers/go.sh (hunk 5 above)
Comment (dmitry-shibanov) [reply images/linux/scripts/installers/go.sh]: Done.

File: images/linux/scripts/installers/go.sh (hunk 2 above)
Comment (dmitry-shibanov) [reply images/linux/scripts/installers/go.sh]: Done.

File: images/linux/scripts/installers/go.sh [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+        echo "GOROOT=$goFolder" | tee -a /etc/environment
+    fi
+}
+
+function getFullGoVersion () {
+    local pattern="refs/tags/go$1([.0-9]{0,3})$"
+    local query='[.[] | select( .ref | test($pattern))] | .[-1] | .ref'
+    refValue=$(jq --arg pattern "$pattern" "$query" $golangTags)
+    version=$(echo "$refValue" | cut -d '/' -f 3)
+    version=$(echo "${version//\"}") # go1.12.17
```
Comment (al-cheb) [inline images/linux/scripts/installers/go.sh]: Why do we need this command?

**Pull request merged**